pinecone/control/pinecone.py

```diff
@@ -320,6 +327,11 @@ def create_index(
         def _parse_non_empty_args(args: List[Tuple[str, Any]]) -> Dict[str, Any]:
             return {arg_name: val for arg_name, val in args if val is not None}
 
+        if deletion_protection in ["enabled", "disabled"]:
+            dp = DeletionProtection(deletion_protection)
+        else:
+            raise ValueError("deletion_protection must be either 'enabled' or 'disabled'")
```

Comment:
The generated `DeletionProtection` class implements a validation, but the error message it emits is very confusing. So I implemented my own check on this here and in the configure method.